File: app/src/main/java/omniex/nl/omniex/ui/app/order/OrderActivity.java
package omniex.nl.omniex.ui.app.order;

import org.androidannotations.annotations.AfterViews;
import org.androidannotations.annotations.EActivity;

import nl.omniex.omniexshopping.R;
import nl.omniex.omniexshopping.data.model.address.Address;
import nl.omniex.omniexshopping.data.model.cart.Cart;
import nl.omniex.omniexshopping.data.model.order.OrderOverview;
import nl.omniex.omniexshopping.data.model.shipping.ShippingMethod;
import nl.omniex.omniexshopping.ui.app.order.shipping.OrderShippingFragment_;
import nl.omniex.omniexshopping.ui.base.BaseActivity;

@EActivity(R.layout.activity_order)
public class OrderActivity extends BaseActivity<OrderView, OrderPresenter> implements OrderView {

    private OrderOverview mOrderOverview = new OrderOverview();

    @Override
    protected void onFirstCreate() {
        super.onFirstCreate();
        getCustomToolbar()
                .setIconStart(R.drawable.twotone_arrow_back_black_36)
                .setIconStarClickListener(this::finish);
    }

    @AfterViews
    void initOrderProcess() {
        getPresenter().getCart();
    }

    @Override
    public void onCartFetched(Cart cart) {
        goToFragment(OrderShippingFragment_.builder().build(), true, "");
        mOrderOverview.setCart(cart);
    }

    public void setShippingAddress(Address shippingAddress){
        mOrderOverview.setShippingAddress(shippingAddress);
    }

    public void setPaymentAddress(Address paymentAddress){
        mOrderOverview.setPaymentAddress(paymentAddress);
    }

    public void setShippingMethod(ShippingMethod shippingMethod){
        mOrderOverview.setShippingMethod(shippingMethod);
    }

    public void setPaymentMethod(String paymentMethod){
        mOrderOverview.setPaymentMethod(paymentMethod);
    }

    public OrderOverview getOrderOverview() {
        return mOrderOverview;
    }

    @Override
    public void startLoading() {
        showProgressBar();
    }

    @Override
    public void stopLoading() {
        hideProgressBar();
    }

    @Override
    public void onOrderOverviewSuccess() {
        goToFragment(OrderShippingFragment_.builder().build(), true, "");
    }

    @Override
    public void onOrderOverviewFailed() {
        finish();
    }
}

File: app/src/main/java/omniex/nl/omniex/ui/app/order/OrderActivity.kt
package omniex.nl.omniex.ui.app.order

import com.hannesdorfmann.mosby3.PresenterManager.getPresenter
import omniex.nl.omniex.R
import omniex.nl.omniex.data.model.address.Address
import omniex.nl.omniex.data.model.cart.Cart
import omniex.nl.omniex.data.model.order.OrderOverview
import omniex.nl.omniex.data.model.shipping.ShippingMethod
import omniex.nl.omniex.ui.base.BaseActivity
import omniex.nl.omniex.ui.views.toolbar.CustomToolbar
import org.androidannotations.annotations.AfterViews
import org.androidannotations.annotations.EActivity

@EActivity(R.layout.activity_order)
open class OrderActivity : BaseActivity<OrderView, OrderPresenter>(), OrderView {

    val orderOverview = OrderOverview()

    override fun onFirstCreate() {
        super.onFirstCreate()
        customToolbar.setIconStart(R.drawable.twotone_arrow_back_black_36)
                .setIconStarClickListener(object : CustomToolbar.IconStartClickListener {
                    override fun onIconStartClick() {
                        finishActivity(1)
                    }
                })
    }

    @AfterViews
    internal fun initOrderProcess() {
        getPresenter().getCart()
    }

    override fun onCartFetched(cart: Cart) {
        goToFragment(OrderShippingFragment_.builder().build(), true, "")
        orderOverview.cart = cart
    }

    fun setShippingAddress(shippingAddress: Address) {
        orderOverview.shippingAddress = shippingAddress
    }

    fun setPaymentAddress(paymentAddress: Address) {
        orderOverview.paymentAddress = paymentAddress
    }

    fun setShippingMethod(shippingMethod: ShippingMethod) {
        orderOverview.shippingMethod = shippingMethod
    }

    fun setPaymentMethod(paymentMethod: String) {
        orderOverview.paymentMethod = paymentMethod
    }

    override fun startLoading() {
        showProgressBar()
    }

    override fun stopLoading() {
        hideProgressBar()
    }

    override fun onOrderOverviewSuccess() {
        goToFragment(OrderShippingFragment_.builder().build(), true, "")
    }

    override fun onOrderOverviewFailed() {
        finish()
    }
}
